oresat_configs/scripts/gen_kaitai.py

```diff
@@ -0,0 +1,226 @@
+"""Generate KaiTai for the beacon."""
+
+from argparse import ArgumentParser, Namespace
+from typing import Any, Optional, cast
+
+import canopen
+from yaml import dump
+
+from .. import Consts, OreSatConfig
+
+GEN_KAITAI = "generate beacon kaitai configuration"
+
+
+def build_parser(parser: ArgumentParser) -> ArgumentParser:
+    """Configures an ArgumentParser suitable for this script.
+
+    The given parser may be standalone or it may be used as a subcommand in another ArgumentParser.
+    """
+    parser.description = GEN_KAITAI
+    parser.add_argument(
+        "--oresat",
+        default=Consts.default().arg,
+        choices=[m.arg for m in Consts],
+        type=lambda x: x.lower().removeprefix("oresat"),
+        help="Oresat Mission. (Default: %(default)s)",
+    )
+    parser.add_argument(
+        "-d", "--dir-path", default=".", help="Output directory path. (Default: %(default)s)"
+    )
+    return parser
+
+
+def register_subparser(subparsers: Any) -> None:
+    """Registers an ArgumentParser as a subcommand of another parser.
+
+    Intended to be called by __main__.py for each script. Given the output of add_subparsers(),
+    (which I think is a subparser group, but is technically unspecified) this function should
+    create its own ArgumentParser via add_parser(). It must also set_default() the func argument
+    to designate the entry point into this script.
+    See https://docs.python.org/3/library/argparse.html#sub-commands, especially the end of that
+    section, for more.
+    """
+    parser = build_parser(subparsers.add_parser("xtce", help=GEN_KAITAI))
+    parser.set_defaults(func=GEN_KAITAI)
+
+
+CANOPEN_TO_KAITAI_DT = {
+    canopen.objectdictionary.BOOLEAN: "s",
+    canopen.objectdictionary.INTEGER8: "int8",
+    canopen.objectdictionary.INTEGER16: "i1",
+    canopen.objectdictionary.INTEGER32: "i2",
+    canopen.objectdictionary.INTEGER64: "i3",
+    canopen.objectdictionary.UNSIGNED8: "u1",
+    canopen.objectdictionary.UNSIGNED16: "u2",
+    canopen.objectdictionary.UNSIGNED32: "u3",
+    canopen.objectdictionary.UNSIGNED64: "u4",
+    canopen.objectdictionary.VISIBLE_STRING: "str",
+    canopen.objectdictionary.REAL32: "float",
+    canopen.objectdictionary.REAL64: "double",
+}
+
+
+def write_kaitai(config: OreSatConfig, dir_path: str = ".") -> None:
+    """Write beacon configs to a kaitai file."""
+
+    # Grab and format mission name
+    name = config.mission.name.lower().replace("_", ".")
+
+    #  Setup pre-determined canned types
+    kaitai_data = {
+        "meta": {
+            "id": name,
+            "title": f"{name} Decoder Struct",
+            "endian": "le",
+        },
+        "doc": "",
```

Comment:
Apparently `ksc` doesn't like empty docstrings. It fails with the very helpful error:
```
oresat0_5: /:
	error: next on empty iterator
```
when this string is empty.

Reply:
removed